integration/images/ruby/3.0/Dockerfile

```diff
@@ -0,0 +1,55 @@
+# https://hub.docker.com/layers/ruby/library/ruby/3.0.0/images/sha256-9719b7df964602b114a5e9319bd0a3ce2d4d44ccb7ea4d1702fa1ef9742a8f84
+FROM ruby@sha256:9719b7df964602b114a5e9319bd0a3ce2d4d44ccb7ea4d1702fa1ef9742a8f84
```

Comment:
I'm curious about this -- is/was the `3.0` image busted?

Reply:
It was; @marcotc has more details but there was some bug in the Ruby core at the time. It may be fixed now... I would like to go to `ruby:3.0` if possible.

Reply:
Because docker maintainers can change the content of what a label represents (e.g. `ruby:3.0` can return image `sha256:9719b7df964602b114a5e9319bd0a3ce2d4d44ccb7ea4d1702fa1ef9742a8f84` today, but a different one in the future), I locked them to the exact image used at their creation time.

This is not strictly necessary, but could help with repeatability of images if we happen to have to tweak them later.

I don't think this is a huge issue, so I think it's fine to change this to `ruby:3.0`. The few times I've seen it happen, small bug fixes were introduced to the image environment, so there was no harm done.

Reply:
Gotcha, I must have been thinking of something of when Ruby 3.0 was in beta.

If this is just a matter of fixing the version, I think its better we pin to `3.0` because we want to run the image other Ruby users would. It's probably a good thing(TM) that CI breaks if an updated `3.0` tag doesn't work with our build, because it'll let us know something might be wrong on our end.